Tasks/DeleteFilesV1/deletefiles.ts

```diff
@@ -85,6 +87,22 @@ tl.setResourcePath(path.join(__dirname, 'task.json'));
         }
     }
 
+    // if there wasn't an error, check if there's anything in the folder tree other than folders
+    // if not, delete the root as well
+    if (removeSourceFolder && !errorHappened) {
+        foundPaths = tl.find(sourceFolder);
+
+        if (foundPaths.every(x => tl.stats(x).isDirectory())) {
```

Comment:
Should we check count === 1 and just check that item?

Reply:
I might just be being overly cautious. I'll run some more tests.